src/actions/courseActions.js

```diff
@@ -0,0 +1,55 @@
+import * as types from './actionTypes';
+import courseApi from '../api/mockCourseApi';
+import { beginAjaxCall, ajaxCallError } from './ajaxStatusActions';
+
+export function loadCoursesSuccess(courses) {
+  return { type: types.LOAD_COURSES_SUCCESS, courses };
+}
+
+export function createCourseSuccess(course) {
+  return { type: types.CREATE_COURSES_SUCCESS, course };
+}
+
+export function updateCourseSuccess(course) {
+  return { type: types.UPDATE_COURSES_SUCCESS, course };
+}
+
+export function deleteCourseSuccess(courseId) {
+  return { type: types.DELETE_COURSES_SUCCESS, courseId };
+}
+
+export function loadCourses() {
+  return function(dispatch) {
+    dispatch(beginAjaxCall());
+    return courseApi.getAllCourses().then(courses => {
+      dispatch(loadCoursesSuccess(courses));
+    }).catch(error => {
+      throw(error);
+    });
+  };
+}
+
+export function saveCourse(course) {
+  return function(dispatch, getState) {
+    dispatch(beginAjaxCall());
+    return courseApi.saveCourse(course).then(savedCourse => {
```

Comment:
I mentioned something like this above. I think the function name should define what is been executed
saveOrUpdateCourse might make more sense